openwisp_controller/config/static/config/js/widget.js

```diff
@@ -153,6 +153,28 @@
         return error;
     };
 
+    var addMaxlengthClass = function() {
+        $('.jsoneditor input[maxlength]:not(.has-max-length)').addClass('has-max-length');
+    };
+    
+    var validateMaxLength = function() {
+        addMaxlengthClass();
+        var fields = $(".has-max-length");
+        fields.map((index, field) => {
+            if ($(field).is(":visible")) {
+                var maxLength = $(field).attr('maxLength');
+                $(field).off('input');
+                $(field).on('input', () => {
+                    if ($(field).val().indexOf('{{') > -1) {
+                        $(field).removeAttr('maxLength');
+                    } else {
+                        $(field).attr('maxLength', maxLength);
+                    }
+                });
+            }
+        });
```

Comment:
I tried something simple which does not need to be called on every change event of the editor.

It seems it also fixes the paste problem. Please check it and improve it / adapt it if needed

```javascript
$('.jsoneditor').delegate('.has-max-length:visible', 'input paste', function(e){
    var field = $(e.target),
        value;

    if (e.originalEvent.type === 'paste') {
        value = e.originalEvent.clipboardData.getData('text');
    } else {
        value = field.val();
    }

    if (value.indexOf('{{') > -1) {
        field.removeAttr('maxlength');
    } else {
        // WARNING: this assumes you set the a data attribute
        // with the maxlength value before
      
        field.attr('maxlength', field.data('maxlength'));
    }
});
```

Do not put this code in the change event of the editor though.
On the change event of the editor the only thing you'd have to do is to add the max length class and data attribute if needed.

**PS**: please always use single quotes or try to be consistent with the quoting style used (although while editing this file I noticed the quoting style is mixed, let's stick to single quotes).